autorest/responder.go

```diff
@@ -166,16 +166,23 @@ func ByUnmarshallingXML(v interface{}) RespondDecorator {
 
 // WithErrorUnlessStatusCode returns a RespondDecorator that emits an error unless the response
 // StatusCode is among the set passed. Since these are artificial errors, the response body
-// may still require closing.
+// may still require closing. On error, the response body is copied and replaced.
```

Comment:
You should probably say, 

> On error, response body is fully read into a buffer and presented in the returned error, as well as in the response body.

Also 

> Since these are artificial errors, the response body may still require closing.

doesn't hold true as we now read the response body "always" when it is available.